integrations/dashly/lib/index.js

```diff
@@ -0,0 +1,223 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+var integration = require('@segment/analytics.js-integration');
+
+/**
+ * Expose `Dashly` integration.
+ */
+
+var Dashly = (module.exports = integration('Dashly')
+  .assumesPageview()
+  .global('dashly')
+  .global('dashlyasync')
+  .option('apiKey', null)
+  .tag('<script src="//cdn.dashly.app/api.min.js"></script>'));
+
+/**
+ * Initialize.
+ *
+ * https://dashly.io/developers/tuning/install/
+ *
+ * @api public
+ */
+
+Dashly.prototype.initialize = function() {
+  var apiKey = this.options.apiKey;
+
+  window.dashly = {};
+  window.dashlyasync = [];
+
+  var m = [
+    'addCallback',
+    'auth',
+    'connect',
+    'identify',
+    'onReady',
+    'open',
+    'removeCallback',
+    'track',
+    'trackMessageInteraction'
+  ];
+
+  for (var i = 0; i < m.length; i++) {
+    window.dashly[m[i]] = Build(m[i]);
+  }
+
+  window.dashly.connect(apiKey);
+
+  this.load(this.ready);
+
+  function Build(name) {
```

Comment:
Nit: I would rather add comment, that this function is a stub, that plans method execution on script loaded and initialized.